app/controllers/admin/schedule_controller.rb

```diff
@@ -0,0 +1,54 @@
+module Admin
+  class ScheduleController < Admin::BaseController
+    # By authorizing 'conference' resource, we can ensure there will be no unauthorized access to
+    # the schedule of a conference, which should not be accessed in the first place
+    load_and_authorize_resource :schedule
```

Comment:
Does it work? I think it should something like 
``
load_and_authorize_resource :schedule, through: :program
``